contracts/staking/stakeManager/StakeManager.sol

```diff
@@ -165,14 +170,22 @@ contract StakeManager is IStakeManager, StakeManagerStorage {
                 ValidatorShare(_contract).activeAmount()
             );
         }
+
         perceivedStake = Math.max(
             perceivedStake,
             validatorAuction[validatorId].amount
         );
+        
+        uint256 finalBid = bid;
+        
+        if (senderValidatorId != 0) {
+            // if current validator bids, increase bid by current stake
+            finalBid = perceivedStake.add(bid);
```

Comment:
What will happen when someone overbids a validator's bid? Will validator be able to instantly withdraw whole stake or just extra bid amount?

Reply:
Whole stake.

Reply:
Realized there is not such test case, adding